pkg/vgmanager/vgmanager_controller.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vgmanager
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	"github.com/red-hat-storage/lvm-operator/pkg/internal"
+	"github.com/topolvm/topolvm"
+	"github.com/topolvm/topolvm/lvmd"
+	lvmdCMD "github.com/topolvm/topolvm/pkg/lvmd/cmd"
+	"gopkg.in/yaml.v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	corev1helper "k8s.io/component-helpers/scheduling/corev1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	ControllerName = "vg-manager"
+)
+
+var ConfigFilePath = "/mnt/lvmd/lvmd.yaml"
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *VGReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.deviceAgeMap = newAgeMap(&wallTime{})
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&lvmv1alpha1.LVMCluster{}).
+		Complete(r)
+}
+
+type VGReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+	// map from KNAME of device to time when the device was first observed since the process started
+	deviceAgeMap *ageMap
+}
+
+func (r *VGReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.Log = log.FromContext(ctx).WithName(ControllerName)
+	r.Log.Info("reconciling", "lvmcluster", req)
+	res, err := r.reconcile(ctx, req)
+	if err != nil {
+		r.Log.Error(err, "reconcile error")
+	}
+	r.Log.V(1).Info("reconcile complete", "result", res)
```

Comment:
What is r.Log.V(1)?

Reply:
Not sure. Looks like different verbosity levels. I'll remove it.